src/orchard/java/parser.clj

```diff
@@ -285,6 +286,18 @@
         (str module "/" path)
         path))))
 
+(def url-protocol (memfn ^URL getProtocol))
+
+(defn jar-path
+  "Given a jar:file:...!/... URL, return the location of the jar file on the
+  filesystem. Returns nil on any other URL."
+  [^URL jar-resource]
+  (when (= "jar" (url-protocol jar-resource))
```

Comment:
```clj
(assert (= "jar" (url-protocol jar-resource))
  ;; debug info:
  (url-protocol jar-resource))
```

is just as cheap, but also ensures our assumptions hold true. Specifically, `jar-path` has only one caller, so it's pretty easy to reason about what our preconditions are